nautobot/ui/src/components/Retrieve/RenderHeader.js

```diff
@@ -59,24 +58,50 @@ export default function RenderHeader({ data }) {
                 )}
             </Heading>
             <ButtonGroup alignItems="center">
-                <UIButton
-                    size="sm"
-                    variant="primaryAction"
-                    leftIcon={<MeatballsIcon />}
-                    onClick={onOpen}
-                >
-                    Actions
-                </UIButton>
-
-                <Modal isOpen={isOpen} onClose={onClose}>
-                    <ModalOverlay />
-
-                    <ModalContent>
-                        <ModalCloseButton />
-
-                        <ModalBody>To be implemented!</ModalBody>
-                    </ModalContent>
-                </Modal>
+                <Menu>
+                    <MenuButton
+                        as={UIButton}
+                        size="sm"
+                        variant="primaryAction"
+                        leftIcon={<MeatballsIcon />}
+                    >
+                        Actions
+                    </MenuButton>
+                    <MenuList>
+                        <MenuItem
+                            to={`${window.location.pathname}edit/`}
+                            icon={<EditIcon />}
+                            onClick={(e) => {
+                                e.preventDefault();
+                                // Because there is currently no support for Edit view in the new UI for production,
+                                // the code below checks if the app is running in production and redirects the user to
+                                // the Edit page; after the page is reloaded, nautobot takes care of rendering the legacy UI.
+                                if (process.env.NODE_ENV === "production") {
+                                    document.location.href += "edit/";
+                                }
+                            }}
+                        >
+                            {" "}
+                            Edit
+                        </MenuItem>
+                        <MenuItem
+                            to={`${window.location.pathname}delete/`}
+                            icon={<BinIcon />}
+                            onClick={(e) => {
+                                e.preventDefault();
+                                // Because there is currently no support for Delete view in the new UI for production,
+                                // the code below checks if the app is running in production and redirects the user to
+                                // the Delete page; after the page is reloaded, nautobot takes care of rendering the legacy UI.
+                                if (process.env.NODE_ENV === "production") {
+                                    document.location.href += "delete/";
+                                }
```

Comment:
Same here, please: Add a TODO comment. Since we don't plan to keep this gate indefinitely, it will make it easier to find later!